great_expectations/rule_based_profiler/data_assistant/data_assistant_dispatcher.py

```diff
@@ -26,20 +26,20 @@ def __init__(self, data_context: "BaseDataContext") -> None:  # noqa: F821
 
         self._data_assistant_runner_cache = {}
 
-    def __getattr__(self, name) -> DataAssistantRunner:
+    def __getattr__(self, name: str) -> DataAssistantRunner:
         # Both, registered data_assistant_type and alias name are supported for invocation.
 
-        # Attempt to utilize alias name for invocation first.
-        data_assistant_cls: Optional[Type["DataAssistant"]] = get_data_assistant_impl(
-            data_assistant_type=name.lower()
-        )
-        # If no friendly alias exists, use registered data_assistant_type for invocation.
+        # Attempt to utilize data_assistant_type name for invocation first.
+        data_assistant_cls: Optional[
+            Type["DataAssistant"]
+        ] = DataAssistantDispatcher.get_data_assistant_impl(data_assistant_type=name)
+
+        # If no registered data_assistant_type exists, try aliases for invocation.
         if data_assistant_cls is None:
-            data_assistant_cls = get_data_assistant_impl(
-                data_assistant_type=DataAssistantDispatcher.DATA_ASSISTANT_TYPE_ALIASES.get(
-                    name.lower()
-                )
-            )
+            for type_ in DataAssistantDispatcher._registered_data_assistants.values():
```

Comment:
Less efficient than a direct lookup but we have so few assistants that moving from O(1) to O(N) is negligible.

Reply:
consider flipping the order here; most folks will reference assistants with the name.

... and why not do `x = DataAssistantDispatcher._registered_data_assistants.get(name)`; `if x is None ....`?

Reply:
Although I just mentioned that performance was not an issue, I went with this approach so we could leverage the O(1) lookup of the dictionary and only do the iteration through types if needed.

I can reverse this! In regards to `get`, that definitely works. Funny how I missed that after we just talked about that pattern earlier today 😆